# Analisis Profiling Forensik Tesis
### Nama: Fikri Irfan Adristi
### NIM: 23917020
### Program Studi Informatika Progam Magister Universitas Islam Indonesia

# Import Packages & Load Dataset

In [ ]:
# Import Packages
from google.colab import drive
import pandas as pd

# Mount Google Drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [ ]:
# Load datasets from Google Drive
altenen_data = pd.read_excel('/content/gdrive/MyDrive/Tesis Fikri/Dataset Tesis/Data Altenen Porn Section.xlsx')
astradumps_data = pd.read_excel('/content/gdrive/MyDrive/Tesis Fikri/Dataset Tesis/Data Astradumps.xlsx')
cracking_tutorial_data = pd.read_excel('/content/gdrive/MyDrive/Tesis Fikri/Dataset Tesis/Data Carding.Store Cracking Tutorial Section.xlsx')
money_heist = pd.read_excel('/content/gdrive/MyDrive/Tesis Fikri/Dataset Tesis/Data Money-Heist.xlsx')

# Inisialisasi Scripting

In [ ]:
# Data Pre-processing Viewer
def data_preprocessing_viewer(data, dataset_name):
    print(f"\n=== {dataset_name} ===")
    print("\nInfo:")
    print(data.info())
    print("\nMissing Values:")
    print(data.isnull().sum())
    print("\nDuplicated Rows:")
    print(data[data.duplicated()])

In [ ]:
# Profiling Altenen Porn Section
def profile_altenen(data):
    print("\n=== Altenen Porn Section ===")
    print("Columns:", data.columns)
    print("Summary:\n", data.describe(include='all'))
    print("Top Thread Writers:", data['Thread Writer Member'].value_counts().head(5))
    print("Most Active Threads:", data[['Thread', 'Replies', 'Views']].sort_values(by=['Replies', 'Views'], ascending=False).head(5))

In [ ]:
# Profiling Astradumps
def profile_astradumps(data):
    print("\n=== Astradumps ===")
    print("Columns:", data.columns)
    print("Summary:\n", data.describe(include='all'))
    print("Top Categories:", data['Product Categories'].value_counts().head(5))
    print("Most Expensive Items:\n", data.sort_values(by='Price', ascending=False).head(5))

In [ ]:
# Profiling Carding.Store Cracking Tutorial Section
def profile_cracking_tutorial(data):
    print("\n=== Cracking Tutorial Section ===")
    print("Columns:", data.columns)
    print("Summary:\n", data.describe(include='all'))
    print("Top Thread Writers:", data['First Thread Writer'].value_counts().head(5))
    print("Most Viewed Threads:\n", data[['Thread Title', 'Views']].sort_values(by='Views', ascending=False).head(5))

In [ ]:
# Profiling Money-Heist
def profile_money_heist(data):
    print("\n=== Money-Heist ===")
    print("Columns:", data.columns)  # Untuk memastikan kolom yang benar
    print("Summary:\n", data.describe(include='all'))  # Menampilkan ringkasan
    print("Top Categories:", data['Categories'].value_counts().head(5))  # Top 5 kategori
    print("Most Expensive Items:\n", data.sort_values(by='Price', ascending=False).head(5))  # Item termahal

# Melakukan Pre-preprocessing untuk Setiap Dataset

In [ ]:
data_preprocessing_viewer(altenen_data, "Altenen Porn Section")


=== Altenen Porn Section ===

Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14391 entries, 0 to 14390
Data columns (total 6 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Thread                14387 non-null  object
 1   Rating & Votes        14391 non-null  object
 2   Replies               14391 non-null  object
 3   Views                 14391 non-null  object
 4   Time Published        14391 non-null  object
 5   Thread Writer Member  14391 non-null  object
dtypes: object(6)
memory usage: 674.7+ KB
None

Missing Values:
Thread                  4
Rating & Votes          0
Replies                 0
Views                   0
Time Published          0
Thread Writer Member    0
dtype: int64

Duplicated Rows:
                                                  Thread Rating & Votes  \
20            Statewins collection snapchat hacked drops                  
2117   ??[Porn LEAK] ?? [K2S / TZ] Camshow – webcamre.

Dataset Altenen Porn Section berisi 14.387 entri untuk kolom *Thread* dan 14.391 entri untuk 5 kolom sisanya, yaitu *Rating & Votes, Replies, Views, Time Published, *dan *Thread Writer Member*. Kolom *Thread* memiliki 4 nilai yang hilang, namun hal ini tidak mempengaruhi kualitas analisis karena lebih fokus pada pemrosesan teks dengan teknik NLP LDA, yang dapat menangani kekosongan data dalam konteks tersebut. Semua kolom lainnya tidak memiliki nilai yang hilang, menunjukkan kelengkapan data yang baik.

Kolom R*ating & Votes, Replies,* dan *Views* bertipe *object*, yang menunjukkan bahwa data tersebut berbentuk string. Ini berarti bahwa data tersebut sudah dalam format yang sesuai dengan cara penyimpanan atau tampilan, seperti "5 stars" atau "100 votes". Oleh karena itu, tipe data ini tidak dapat digunakan untuk analisis infografis. Kolom *Time Published* berisi informasi waktu publikasi dan juga bertipe *object*, namun hal ini dapat tetap dipertahankan dalam format tersebut jika fokus analisis tidak membutuhkan pengelolaan waktu yang lebih mendalam. Kolom *Thread Writer Member* memiliki data yang lengkap dan juga bertipe *object*, yang menunjukkan setiap thread memiliki penulis yang tercatat secara jelas.

Data duplikat mencerminkan strategi pengguna meningkatkan visibilitas thread di forum komunitas. Namun, hal ini tidak menjadi masalah karena tetap mendukung analisis LDA terhadap teks pada thread tersebut. Secara keseluruhan, dataset ini memiliki struktur yang baik dan siap untuk analisis lebih lanjut tanpa memerlukan perubahan besar pada tipe data kolom-kolom tertentu.

In [ ]:
data_preprocessing_viewer(astradumps_data, "Astradumps")


=== Astradumps ===

Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Product Categories  400 non-null    object
 1   Name of Item Sold   400 non-null    object
 2   Price               400 non-null    int64 
 3   Description         400 non-null    object
dtypes: int64(1), object(3)
memory usage: 12.6+ KB
None

Missing Values:
Product Categories    0
Name of Item Sold     0
Price                 0
Description           0
dtype: int64

Duplicated Rows:
    Product Categories                      Name of Item Sold  Price  \
216  Carded Gift Cards  $700 AUD Amazon Gift Card – AUSTRALIA    150   

                                           Description  
216  Description\n$700 AUD Amazon Gift Card – AUSTR...  


*Dataset* yang dianalisis terdiri dari 400 baris dan 4 kolom, mencakup informasi terkait kategori produk *(Product Categories)*, nama item yang dijual *(Name of Item Sold)*, harga *(Price)*, dan deskripsi produk *(Description)*. Kolom *Price* bertipe *integer* (int64), sementara kolom lainnya, seperti *Product Categories, Name of Item Sold,* dan *Description*, bertipe objek *(object)*, yang berisi data berupa string.

Tidak ditemukan nilai yang hilang *(missing values)* dalam *dataset* ini, yang berarti data ini lengkap dan siap untuk analisis lebih lanjut. *Dataset* ini juga memiliki ukuran memori sekitar 12.6 KB, menunjukkan bahwa ukuran data relatif kecil dan dapat dengan mudah diproses dalam memori komputer.

Dalam hal duplikasi, baris ke-216 menunjukkan adanya entri yang sama untuk kategori produk "Carded Gift Cards" dengan nama item "$700 AUD Amazon Gift Card – AUSTRALIA" dan harga 150. Hal ini mencerminkan fakta bahwa produk yang sama dapat terdaftar lebih dari sekali di website Astradumps. Karena produk ini memang muncul lebih dari satu kali dalam platform tersebut, data duplikat ini dianggap valid dan tidak perlu dihapus.

Secara keseluruhan, dataset ini menyajikan informasi yang lengkap dan representatif mengenai produk yang dijual di website tersebut, dengan kolom bertipe objek yang dapat digunakan untuk analisis berbasis teks seperti *Latent Dirichlet Allocation* (LDA) untuk identifikasi pola dan topik, sementara kolom harga yang bertipe *integer* dapat dipakai untuk membuat infografis terkait distribusi harga produk yang dijual.

In [ ]:
data_preprocessing_viewer(cracking_tutorial_data, "Cracking Tutorial Section")


=== Cracking Tutorial Section ===

Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1030 entries, 0 to 1029
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Thread Title          1030 non-null   object
 1   First Thread Writer   1030 non-null   object
 2   Replies               1030 non-null   int64 
 3   Views                 1030 non-null   int64 
 4   Second Thread Writer  1030 non-null   object
dtypes: int64(2), object(3)
memory usage: 40.4+ KB
None

Missing Values:
Thread Title            0
First Thread Writer     0
Replies                 0
Views                   0
Second Thread Writer    0
dtype: int64

Duplicated Rows:
Empty DataFrame
Columns: [Thread Title, First Thread Writer, Replies, Views, Second Thread Writer]
Index: []


*Dataset* Cracking Tutorial Section terdiri dari 1.030 entri dengan 5 kolom, yaitu *Thread Title, First Thread Writer, Replies, Views,* dan *Second Thread Writer*. Semua kolom tidak memiliki nilai yang hilang, sehingga dataset ini lengkap dan siap digunakan untuk analisis. Kolom Thread Title berisi judul-judul thread yang dapat dianalisis menggunakan LDA *(Latent Dirichlet Allocation)* untuk mengidentifikasi topik atau tema utama yang dibahas dalam tutorial cracking.

Kolom *First Thread Writer* dan *Second Thread Writer* menyediakan informasi tentang penulis thread, yang dapat dimanfaatkan untuk menganalisis pola kontribusi penulis atau kolaborasi di antara mereka. Kolom *Replies* dan *Views*, yang bertipe data *integer*, dapat dimanfaatkan untuk mengidentifikasi pola diskusi atau popularitas *thread* dalam bentuk infografis.

Dengan dataset yang *clean* dan lengkap, setiap kolom dalam Cracking Tutorial Section dapat mendukung analisis untuk memahami tren diskusi, popularitas topik, serta pola kontribusi komunitas di *platform* tersebut.

In [ ]:
data_preprocessing_viewer(money_heist, "Money-Heist")


=== Money-Heist ===

Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 229 entries, 0 to 228
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Categories         229 non-null    object
 1   Name of Item Sold  229 non-null    object
 2   Price              229 non-null    int64 
 3   Description        229 non-null    object
dtypes: int64(1), object(3)
memory usage: 7.3+ KB
None

Missing Values:
Categories           0
Name of Item Sold    0
Price                0
Description          0
dtype: int64

Duplicated Rows:
Empty DataFrame
Columns: [Categories, Name of Item Sold, Price, Description]
Index: []


*Dataset* Money-Heist terdiri dari 229 baris dan 4 kolom, yang mencakup kategori produk *(Categories)*, nama item yang dijual *(Name of Item Sold)*, harga *(Price)*, dan deskripsi produk *(Description)*. Kolom *Price* bertipe *integer* (int64), sementara kolom lainnya—*Categories, Name of Item Sold,* dan *Description*—bertipe objek *(object)*, yang berisi data berupa *string*.

Tidak ada nilai yang hilang *(missing values)* dalam *dataset* ini, yang menandakan bahwa semua data terisi dengan lengkap dan siap untuk dianalisis lebih lanjut. Ukuran memori yang digunakan oleh *dataset* ini adalah sekitar 7.3 KB, yang mengindikasikan bahwa *dataset* ini relatif kecil dan dapat diproses dengan efisien.

Dalam hal duplikasi, tidak ditemukan baris yang terduplikasi, yang berarti bahwa setiap entri dalam *dataset* ini adalah unik dan tidak ada redundansi data.

Terkait dengan analisis lebih lanjut, kolom *Categories, Name of Item Sold,* dan *Description* yang bertipe objek *(string)* dapat digunakan untuk analisis berbasis teks, seperti *Latent Dirichlet Allocation* (LDA), yang dapat mengidentifikasi pola dan topik tersembunyi dalam data produk yang dijual. Sementara itu, kolom *Price* yang bertipe *integer* dapat dimanfaatkan untuk membuat infografis terkait distribusi harga produk, yang dapat membantu menggambarkan variasi harga berdasarkan kategori atau produk tertentu.

Secara keseluruhan, dataset ini menawarkan informasi yang lengkap dan dapat digunakan untuk berbagai jenis analisis, baik berbasis teks (LDA) untuk menemukan topik atau tren produk yang muncul, maupun berbasis data numerik (infografis) untuk memahami distribusi harga produk yang dijual.

# Melakukan Profiling untuk Setiap Dataset

In [ ]:
profile_altenen(altenen_data)


=== Altenen Porn Section ===
Columns: Index(['Thread', 'Rating & Votes', 'Replies', 'Views', 'Time Published',
       'Thread Writer Member'],
      dtype='object')
Summary:
                                                    Thread Rating & Votes  \
count                                               14387          14391   
unique                                              13390            481   
top     ⭐[MEGA]⭐⎝❤️REAL GIRLS PACK LEAK GIRLS PRIVATE ...                  
freq                                                   18          10907   

           Replies      Views Time Published Thread Writer Member  
count        14391      14391          14391                14391  
unique         717        818           1826                 5693  
top     Replies\n0  Views\n1K   Dec 15, 2022         GhostlyGamer  
freq          2266       1010             94                  332  
Top Thread Writers: Thread Writer Member
GhostlyGamer     332
Ninjavu          264
huju66           241

Hasil analisis menunjukkan bahwa kolom Thread memiliki tingkat keberagaman yang tinggi dengan 13.390 entri unik dari 14.387 total thread. Thread dengan judul menarik seperti "⭐[MEGA]⭐⎝❤️REAL GIRLS PACK LEAK GIRLS PRIVATE..." adalah yang paling sering muncul.

Kolom Rating & Votes sebagian besar kosong atau tidak digunakan secara informatif, karena lebih dari 75% entri berasal dari kategori "kosong."

Kolom Replies dan Views didominasi oleh beberapa entri populer seperti "Replies 997" dan "Views 3K." Meski ada variasi, pola aktivitas menunjukkan preferensi terhadap konten yang eksplisit dan bersifat "sensasi."

Penulis thread terbanyak adalah GhostlyGamer dengan kontribusi 332 thread, diikuti oleh Ninjavu (264) dan lainnya. Adanya thread yang sangat aktif seperti "Busty Blonde Milf Gets Railed" menunjukkan korelasi tinggi antara jumlah thread populer dan topik tertentu.

*Thread* populer memiliki jumlah balasan dan jumlah tampilan yang signifikan, menjadikannya pusat perhatian pengguna. Beberapa judul *thread* bahkan menunjukkan upaya eksplisit dalam menarik audiens dengan gaya *clickbait*.

Secara keseluruhan, hasil ini menyoroti pola kontribusi komunitas dalam kategori ini yang bisa dimanfaatkan untuk analisis lebih lanjut dengan *Natural Language Processing* (NLP) guna mengungkap pola, tren, dan potensi aktivitas ilegal di forum ini.

In [ ]:
profile_astradumps(astradumps_data)


=== Astradumps ===
Columns: Index(['Product Categories', 'Name of Item Sold', 'Price', 'Description'], dtype='object')
Summary:
        Product Categories                                  Name of Item Sold  \
count                 400                                                400   
unique                 24                                                392   
top           Bank Logins  USA NON-VBV CARDS (CREDIT/DEBIT CARD) + FREE C...   
freq                  103                                                  2   
mean                  NaN                                                NaN   
std                   NaN                                                NaN   
min                   NaN                                                NaN   
25%                   NaN                                                NaN   
50%                   NaN                                                NaN   
75%                   NaN                                             

Dataset Astradumps terdiri dari 400 baris dan 4 kolom yang mencakup kategori produk *(Product Categories)*, nama item yang dijual *(Name of Item Sold)*, harga *(Price)*, dan deskripsi produk *(Description)*. Kolom *Price* memiliki nilai numerik, dengan harga rata-rata sebesar 863,83; yang menunjukkan bahwa harga produk bervariasi cukup luas, dengan harga terendah sebesar 0 dan harga tertinggi mencapai 45.700. Kolom *Product Categories* berisi 24 kategori unik, dengan kategori yang paling banyak muncul adalah "*Bank Logins*" yang ditemukan 103 kali, diikuti oleh "*Accounts*" sebanyak 62 kali dan "*CVV and Cards*" sebanyak 44 kali.

Kolom *Name of Item Sold* mencatat 392 item unik, dengan item yang paling sering muncul adalah "USA NON-VBV CARDS (CREDIT/DEBIT CARD) + FREE C...", yang ditemukan 2 kali. Kolom *Description* juga mencatat 398 deskripsi unik, dengan deskripsi yang paling sering muncul berkaitan dengan "NEWLY ITALY CC + CVV + INFO $50K". Dataset ini tidak memiliki nilai yang hilang, sehingga dapat langsung digunakan untuk analisis lebih lanjut.

Terkait dengan harga produk, beberapa produk dengan harga tertinggi ditemukan, seperti "Get $500000 Bank Transfer with Instant Credit" yang dihargai 45.700, "Buy $200000 Cloned ATM card with Balance" seharga 10.000, dan produk lain yang mencakup layanan transfer uang dan kartu ATM dengan harga tinggi. Dengan data harga ini, kita bisa mengidentifikasi produk-produk premium yang mungkin menarik untuk analisis lebih mendalam, termasuk pola harga dalam kategori tertentu.

Secara keseluruhan, dataset ini memberikan gambaran yang jelas tentang produk yang dijual di website Astradumps, baik dari segi kategori, harga, maupun deskripsi produk, yang bisa digunakan untuk analisis lebih lanjut mengenai tren penjualan dan harga.

In [ ]:
profile_cracking_tutorial(cracking_tutorial_data)


=== Cracking Tutorial Section ===
Columns: Index(['Thread Title', 'First Thread Writer', 'Replies', 'Views',
       'Second Thread Writer'],
      dtype='object')
Summary:
        Thread Title First Thread Writer      Replies         Views  \
count          1030                1030  1030.000000   1030.000000   
unique          999                 472          NaN           NaN   
top       [supreme]             JuNaiD™          NaN           NaN   
freq             16                  43          NaN           NaN   
mean            NaN                 NaN    73.105825   3802.318447   
std             NaN                 NaN   185.255135   7421.333137   
min             NaN                 NaN     0.000000    522.000000   
25%             NaN                 NaN     3.000000    832.000000   
50%             NaN                 NaN    19.000000   1409.000000   
75%             NaN                 NaN    70.000000   3747.000000   
max             NaN                 NaN  2630.000000  98

*Dataset* Cracking Tutorial Section mencakup 1.030 entri yang terdiri dari lima kolom: judul thread *(Thread Title)*, penulis *thread* pertama *(First Thread Writer)*, jumlah balasan *(Replies)*, jumlah tampilan *(Views)*, dan penulis *thread* kedua *(Second Thread Writer)*. Kolom *Replies* memiliki rata-rata 73,11 balasan per *thread*, dengan variasi yang cukup besar, mulai dari 0 hingga 2.630 balasan. Kolom *Views* menunjukkan rata-rata 3.802 tampilan per *thread*, dengan beberapa *thread* yang mendapatkan tampilan yang sangat tinggi, bahkan mencapai 98.202 tampilan, menunjukkan bahwa topik tertentu memiliki daya tarik yang jauh lebih besar dibandingkan yang lainnya.

Kolom *First Thread Writer* mencatat bahwa penulis yang paling aktif adalah "JuNaiD™", yang menulis 43 *thread*, diikuti oleh "Like it" dengan 41 *thread* dan penulis lainnya dengan jumlah *thread* yang lebih sedikit. Hal ini menunjukkan bahwa beberapa penulis memiliki kontribusi yang lebih besar dalam menciptakan konten dalam forum ini. Kolom *Second Thread Writer* mencatat 883 penulis unik, dengan "JuNaiD™" juga mencatatkan frekuensi tertinggi di kolom ini, yang menunjukkan kontribusi signifikan dari penulis tersebut pada *thread-thread* yang ditulis oleh orang lain.

Dalam analisis ini, kita dapat melihat bahwa ada sejumlah *thread* dengan jumlah tampilan yang sangat tinggi, seperti "Netflix Cracking Tutorial For Beginners (FAST...)" yang mendapatkan 98.202 tampilan, yang dapat menarik perhatian lebih dalam analisis popularitas dan dampak topik yang dibahas. Dengan demikian, informasi ini memberikan gambaran yang jelas tentang dinamika penulisan *thread* dan interaksi dalam forum *tutorial cracking*, yang dapat digunakan untuk mengidentifikasi konten yang paling populer dan penulis yang paling produktif.

In [ ]:
profile_money_heist (money_heist)


=== Money-Heist ===
Columns: Index(['Categories', 'Name of Item Sold', 'Price', 'Description'], dtype='object')
Summary:
                 Categories                                  Name of Item Sold  \
count                  229                                                229   
unique                  27                                                214   
top     Credit/Debit Cards  ACR38 R4 RFID Smart Contact Chip EMV Card Read...   
freq                    46                                                  2   
mean                   NaN                                                NaN   
std                    NaN                                                NaN   
min                    NaN                                                NaN   
25%                    NaN                                                NaN   
50%                    NaN                                                NaN   
75%                    NaN                                         

Dataset Money-Heist berisi 229 entri dengan empat kolom: kategori produk *(Categories)*, nama item yang dijual *(Name of Item Sold)*, harga *(Price)*, dan deskripsi produk *(Description)*. Kolom *Categories* mencatatkan 27 kategori unik, dengan kategori yang paling banyak muncul adalah "Credit/Debit Cards" yang tercatat 46 kali, diikuti oleh kategori "Dumps" (24 kali) dan "Burners" (23 kali). Hal ini menunjukkan bahwa kategori produk tertentu lebih dominan di *platform* ini.

Kolom *Name of Item Sold* mencatatkan 214 nama produk unik, dengan produk yang paling sering muncul adalah "ACR38 R4 RFID Smart Contact Chip EMV Card Reader Writer", yang terjual dua kali. Rata-rata harga produk dalam dataset ini adalah sekitar 316,38; dengan harga yang bervariasi secara signifikan, mulai dari 20 hingga 10.000, menunjukkan adanya produk dengan harga yang sangat tinggi. Kolom *Description* berisi 192 deskripsi unik, dengan satu deskripsi muncul 15 kali, menunjukkan adanya produk dengan penawaran yang hampir serupa namun berbeda dalam konteks atau detail penawaran.

Beberapa item yang paling mahal, seperti "Bitcoin Mixing Service – Worldwide" dengan harga 10.000, atau "GSM DATA RECEIVER + LARGE DISTANCE ANTENNA + EMV" dengan harga 1.500, menyoroti adanya transaksi dengan nilai tinggi yang terkait dengan produk spesifik, terutama yang terkait dengan layanan dan peralatan untuk kegiatan teknis. Analisis ini menggambarkan variasi produk, harga, dan kategori yang dijual dalam platform tersebut, serta potensi minat dan fokus dari para pembeli dan penjual.